### PR TITLE
Add RBE Constraint

### PR DESCRIPTION
**Pull request by ajdubas**
Add RBE Constraint

### CODE REVIEW

File: src/constraints/RBEConstraint.C [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+RBEConstraint::validParams()
+{
+  InputParameters params = NodalConstraint::validParams();
+  params.addClassDescription(
+    "Constrains secondary node to move as a linear combination of primary nodes.");
+  params.addRequiredParam<BoundaryName>("primary_node_set",
+    "The boundary ID associated with the primary nodes.");
+  params.addRequiredParam<BoundaryName>("secondary_node_set",
+    "The boundary ID associated with the secondary side.");
+  params.addRequiredParam<Real>("penalty", "The penalty used for the boundary term.");
```
Comment (lukethehuman) [inline src/constraints/RBEConstraint.C:16]: Is there anything to add about what range of values are valid for the penalty?

**Review by lukethehuman — APPROVED** [review]
Looks good, I've left a few comments on things that weren't immediately clear to me. Since tests/examples are coming in a separate PR, and none of my comments are major blockers, I've marked this as approved.

File: src/constraints/RBEConstraint.C [hunk 2]
```diff
@@ -0,0 +1,107 @@
+#include "RBEConstraint.h"
+#include "MooseMesh.h"
+
+registerMooseObject("ProteusApp", RBEConstraint);
```
[the remark this answers is not in the document]
Comment (ajdubas) [reply src/constraints/RBEConstraint.C:4]: I think we can assume it's a known acronym for anyone using it, but added clarification to the docstring anyway.

File: src/constraints/RBEConstraint.C (hunk 1 above)
Comment (ajdubas) [reply src/constraints/RBEConstraint.C:17]: No, I think we can assume the user knows to use a positive number.

File: src/constraints/RBEConstraint.C [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+InputParameters
+RBEConstraint::validParams()
+{
+  InputParameters params = NodalConstraint::validParams();
+  params.addClassDescription(
+    "Constrains secondary node to move as a linear combination of primary nodes.");
+  params.addRequiredParam<BoundaryName>("primary_node_set",
+    "The boundary ID associated with the primary nodes.");
+  params.addRequiredParam<BoundaryName>("secondary_node_set",
+    "The boundary ID associated with the secondary side.");
```
[the remark this answers is not in the document]
Comment (ajdubas) [reply src/constraints/RBEConstraint.C]: Not a typo but a slight inaccuracy, now fixed.

**Pull request merged**